internal/clients/state/v3/state.go

```diff
@@ -158,5 +160,5 @@ func (c *Client) GetKey(_ context.Context, key string) (string, error) {
 	if val, ok := c.mem[key]; ok {
 		return val, nil
 	}
-	return "", fmt.Errorf("key not found")
+	return "", nil
```

Comment:
I decided to return `""` if not found but we can later on make this an option which the user can decide if they want to use `""` or `NotFoundErr`